pkg/controllers/sandbox/sandbox_controller.go

```diff
@@ -158,6 +158,8 @@ func (r *SandboxConfigMapReconciler) constructActors(vdb *v1.VerticaDB, log logr
 		MakeVerifyDeploymentReconciler(r, vdb, log),
 		// Move the subclusters from a sandbox to the main cluster
 		MakeUnsandboxSubclusterReconciler(r, vdb, log, r.Client, pfacts, dispatcher, configMap),
+		// Update the vertica image for unsandboxed subclusters
+		MakeVerticaImageReconciler(r, vdb, log, pfacts),
```

Comment:
There may be problems if we do this after deleting the ConfigMap. Once the ConfigMap is deleted, the controller that is responsible for the pods has transferred back to the vdb controller. Is it possible to decouple the ConfigMap deletion for unsandbox into a separate reconciler that is called _after_ this reconciler? Or another approach is to work this reconciler into the vdb controller.

Reply:
I prefer this reconciler in vdb controller. After unsandbox, the status are updated so the subcluster is no longer tied to the sandbox and should be handle by vdb controller. Instead of having `pfactsForMainCluster` ant not use the sandbox pfacts, you can simply implement this in the vdb controller.